textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,311 @@
+ *	Literal Tokens
+ */
+TOKEN : { < NUMBER_LITERAL : ("-")? ((["0"-"9"])+ ("." (["0"-"9"])*)? | "." (["0"-"9"])+) > }
+TOKEN : { < STRING_LITERAL : "\"" ( "\\" ~[] | ~["\""] )* "\"" > }
+TOKEN : { < REGEX_LITERAL : "/" ( "\\" ~[] | ~["/"] )* "/" > }
+TOKEN : { < IDENTIFIER : ["a"-"z"] (["a"-"z","0"-"9"])* > }
+
+/*
+ *	Wrappers for Literal Tokens
+ */
+double numberDouble():
+{
+    String numberString;
+}
+{
+    numberString = numberLiteral() { return Double.parseDouble(numberString); }
+}
+int numberInteger():
+{
+    String numberString;
+}
+{
+    numberString = numberLiteral() { return Integer.parseInt(numberString); }
+}
+String numberLiteral():
+{
+    Token token;
+}
+{
+    token = <NUMBER_LITERAL> { return token.image; }
+}
+String regexLiteral():
+{
+    Token token;
+}
+{
+    token = <REGEX_LITERAL> { return token.image.substring(1, token.image.length()-1).replace("\\/", "/"); }
+}
+String stringLiteral():
+{
+    Token token;
+}
+{
+    token = <STRING_LITERAL> { return token.image.substring(1, token.image.length()-1).replace("\\\"", "\""); }
+}
+String identifier():
+{
+    Token token;
+}
+{
+    token = <IDENTIFIER> { return token.image; }
+}
+
+/*
+ *	List of (some) Literal Tokens separated by comma
+ */
+List<String> identifierList():
+{
+    String identif;
+    List<String> identifiers = new ArrayList<String>();
+}
+{
+    identif=identifier() { identifiers.add(identif); }
+    ( <COMMA> identif=identifier() { identifiers.add(identif); } )*
+    { return identifiers; }
+}
+
+
+/*
+ *	Statements Declaration
+ */
+List<Map<String, Object>> statementsMain(Consumer<Map<String,Object>> statementConsumer) : 
+{
+    List<Map<String, Object>> statements = new ArrayList();
+    Map<String, Object> newStatement;
+}
+{
+    (
+        newStatement = statement() { statements.add(newStatement); }
+        { if(statementConsumer!=null)statementConsumer.accept(newStatement); }
+    )*
+    <EOF>
+    { return statements; }
+}
+
+Map<String, Object> statement() : 
+{
+    Map<String, Object> newStatement;
+}
+{
+    (
+            newStatement = selectStatement() 
+        |
+            newStatement = createViewStatement()
+        |
+            newStatement = emptyStatement()
+    )
+    <SEMICOLON>
+    { return newStatement; }
+}
+
+Map<String, Object> selectStatement() : 
+{
+    HashMap<String, Object> statementParameters = new HashMap<String, Object>() {{
+        put("statementType", "select");
+        put("statementName", getNewUniqueId());
+        put("projectAll", null);
+        put("projectFields", null);
+        put("extractCommand", null);
+        put("from", null);
+        put("limit", null);
+        put("offset", null);
+    }};
+    
+    List<String> projectFields;
+    Map<String,Object> extractCmd;
+    String identifier;
+    String from;
+    Integer limit;
+    Integer offset;
+
+}
+{
+    (
+            <SELECT> 
+            ( 
+                    <STAR> { statementParameters.put("projectAll", true);  } 
```

Comment:
Add some comments here along with some examples, so we know what is happening in each of these branches.